draft tuqmano blogpost (#623)
ropensci/roweb3: content/blog/2023-08-24-divide-y-venceras-de-polar-al-polarverse/index.en.md

<img src="https://lh5.googleusercontent.com/L0544tZ7i2883QLp7qQGoix1hqLwjppjd9LYnE95Csh-Sq78J_FjRBYh_GZ6Hu1D0QVBzWf0unNjhTOWwClpNkoSjYoNWUjRtE7sVt5-s8KtOGr646jorwPSIJaPduvDSkJJ-bzpArzD-2HfpHGTsew" width="200">
<img src="https://lh3.googleusercontent.com/JhbZ-qOlqRNW-AVAho4A_xjEVw_NUSwwBPy2fM7DwirTKXclyuJfmPm0puMqv1MxaLCIH6ljZ6yU0pDvH-t4TPZenzjHbekNpChPuSrKiWSNH_cKRqxy4iMDdYztWJnjHNM9q10VJ0JC2VZKDHLEBpY" width="200">

> ## The Birth of {polAr}

Reviewer: ```suggestion
## The birth of polAr
```

> With the aim of presenting at the third edition of LatinR, I developed [`{geofacetAR}`](https://electorarg.github.io/geofaceteAR/) a polished version with functions to work with the design of grids as if they were maps of Argentina.

Reviewer: ```suggestion
With the aim of presenting at the third LatinR conference, I developed a polished version of [geofacetAR](https://electorarg.github.io/geofaceteAR/) with functions to work with grid designs as if they were maps of Argentina.
```

> The Covid19 pandemic hit. Learning how to develop packages, in times of confinement and many hours in front of the computer, turned the electoral data processing work of my life as a political scientist and consultant into a new package: [`{polAr}`](https://github.com/electorArg/polAr); that on May 16, 2020 [traveled to CRAN](https://twitter.com/CRANberriesFeed/status/1261597845808975872).

Reviewer: ```suggestion
The Covid19 pandemic hit. Learning how to develop packages, in times of confinement and many hours in front of the computer, turned the electoral data processing work of my life as a political scientist and consultant into a new package: [polAr](https://github.com/electorArg/polAr); that on May 16, 2020 [traveled to CRAN](https://twitter.com/CRANberriesFeed/status/1261597845808975872).
```

## The polAr Universe

> Although originally the package sought to facilitate access and tools for the analysis of electoral data in Argentina, over time, other workflows and data sources (geographic, presidential speeches, and legislative votes) were integrated, with an increase in dependencies and difficulty in managing more and more functions. This was a relevant issue that [was discussed with colleagues in the Package Development panel during the third edition of LatinR (2020)](https://www.youtube.com/watch?v=UYvSv8StDa8&t=10872s).

Reviewer: ```suggestion
Although originally the package sought to facilitate access and tools for the analysis of electoral data in Argentina, over time other workflows and data sources (geographic, presidential speeches, and legislative votes) were integrated, with an increase in dependencies and difficulty in managing more and more functions. This was a problematic issue that [was discussed with colleagues in the Package Development panel during the third LatinR conference (2020)](https://www.youtube.com/watch?v=UYvSv8StDa8&t=10872s).
```

> Although originally the package sought to facilitate access and tools for the analysis of electoral data in Argentina, over time, other workflows and data sources (geographic, presidential speeches, and legislative votes) were integrated, with an increase in dependencies and difficulty in managing more and more functions. This was a relevant issue that [was discussed with colleagues in the Package Development panel during the third edition of LatinR (2020)](https://www.youtube.com/watch?v=UYvSv8StDa8&t=10872s).

Reviewer: In the above paragraph it sounds as if you developed polAr by yourself, but below you talk a lot about 'we'. Did you add other authors on as you added more functionality to the package? If so, perhaps clarify that here?

> [![](https://lh5.googleusercontent.com/uQA8VbmFABjeZbo96fkAtxOqQCLyYnG-PYpDpyzPrs01VrB2bKZDm2X9-jelYXnBvrENlHJxYGcw_h8FSo2jIREks_fHvkPHRXf0ejlcTNqYjdW4wPKZujS2n4GKrnTFZyabIQNUpRzehqRq1FHlbLY)](https://www.youtube.com/watch?v=UYvSv8StDa8&t=10872s)

Reviewer: As above, also, if the caption could be in the text, that would allow us to have it in the correct language for each post and make it accessible to screen readers!

> `{polAr}` had reached 4000 downloads when I receive an email from the CRAN team informing me that I had to adjust a policy violation so that the package would not be archived. Despite submitting a new version the package was archived for no clear reason.

Reviewer: I've softened the language a bit here, these situations are always so frustrating (I've been there 😭 )!
```suggestion
polAr had reached 4000 downloads when I received an email from the CRAN team informing me that I had to fix a policy violation to prevent the package from being archived. Unfortunately my submitted fix did not appear to correct the problem and the package was archived.
```

Author: Indeed! Buy I ' am Ok with the suggestion. Tks

![](https://lh4.googleusercontent.com/BVfIfOecfmQidGrxg4vcISxDw-Cgpri7cxCZcOnHsiR9HW2qwjTR5SRP_AVR27Gf7KFL2Grby_R0nkcbiPlT5ZPh9-GloKBSOuI1fseNGuYOEvopYUe7bAGIBURhX5TX9G3BT4glKRCrr74cyN40Y38)

> If the conversation at LatinR made us think about splitting into multiple packages as a possibility, the CRAN archiving gave us the opportunity to do so. The decision was to generate independent workflows; a strategy that made it possible to move forward at different paces, making specific decisions for each topic, trying to address it at the same time as part of a whole, emulating the *Tidyverse* paradigm: a similar philosophy and grammar of operation for the [collection of packages](https://tuqmano.ar/2021/06/05/de-polar-al-polarverse/). The goal was to improve the development of that which led to the archiving of `{polAr}`.

Reviewer: ```suggestion
If the conversation at LatinR made us think about splitting polAr into multiple packages as a possibility, the CRAN archiving gave us the opportunity to do so. The decision was to generate independent workflows; a strategy that made it possible to move forward at different paces, making specific decisions for each topic, trying to address it at the same time as part of a whole, emulating the *Tidyverse* paradigm: a similar philosophy and grammar of operation for a [collection of packages](https://tuqmano.ar/2021/06/05/de-polar-al-polarverse/). The goal was to improve the development of that which led to the archiving of polAr.
```

> We presented "[Divide and Conquer: from {polAr} to the polarverse](https://github.com/TuQmano/latinr2021/blob/master/divide_reinaras/divide_reinaras.pdf) at the fourth edition of LatinR, showcasing this new universe with five packages:

Reviewer: ```suggestion
We presented "[Divide and Conquer: from {polAr} to the polarverse](https://github.com/TuQmano/latinr2021/blob/master/divide_reinaras/divide_reinaras.pdf) at the fourth LatinR conference, showcasing this new universe with five packages:
```

> 1. 📦 [`{opinAr}`](https://politicaargentina.r-universe.dev/opinAr): provides access to data on public opinion and government in Argentina based on the Confidence in Government Index published monthly by the Universidad Torcuato Di Tella.

Reviewer: ```suggestion
1. 📦 [opinAr](https://politicaargentina.r-universe.dev/opinAr): provides access to data on public opinion and government in Argentina based on the Confidence in Government Index published monthly by the Universidad Torcuato Di Tella.
```

> 2. 📦 [`{electorAr}:`](https://politicaargentina.r-universe.dev/electorAr) facilitates access to electoral data for Argentina (for different levels and with different spatial aggregation) and functions to compute relevant indicators.

Reviewer: ```suggestion
2. 📦 [electorAr](https://politicaargentina.r-universe.dev/electorAr): facilitates access to electoral data for Argentina (for different levels and with different spatial aggregation) and functions to compute relevant indicators.
```

> 3. 📦 [`{legislAr}`](https://politicaargentina.r-universe.dev/legislAr)`:` allows download disaggregated information of votes in the legislative chambers of Argentina, originating from the project [Decade Voted](https://andytow.com/scripts/disciplina/index-d.html) project by Andy Tow.

Reviewer: ```suggestion
3. 📦 [legislAr](https://politicaargentina.r-universe.dev/legislAr): facilitates downloading disaggregated information of votes in the legislative chambers of Argentina, originating from the [Decade Voted](https://andytow.com/scripts/disciplina/index-d.html) project by Andy Tow.
```

> 4. 📦 [`{discursAr}`](https://politicaargentina.r-universe.dev/discursAr)`:` is designed to obtain data on political speeches. In principle, it provides access to presidential speeches in legislative assemblies at the inauguration of ordinary sessions.

Reviewer: ```suggestion
4. 📦 [discursAr](https://politicaargentina.r-universe.dev/discursAr): is designed to obtain data on political speeches. In principle, it provides access to presidential speeches in legislative assemblies at the inauguration of ordinary sessions.
```

> 5. 📦 [`{geoAr}`](https://politicaargentina.r-universe.dev/geoAr): allows access to data and tools for spatial workflows from within R. One [paper](https://github.com/TuQmano/latinr2023/blob/main/geoAr/geoAr.pdf) on the package was accepted for the [sixth edition of LatinR](https://latin-r.com/).

Reviewer: ```suggestion
5. 📦 [geoAr](https://politicaargentina.r-universe.dev/geoAr): allows access to data and tools for spatial workflows from within R. One [paper](https://github.com/TuQmano/latinr2023/blob/main/geoAr/geoAr.pdf) on the package was accepted for the [sixth LatinR](https://latin-r.com/) conference.
```

> In addition, we work on an auxiliary library called [`{polarViz}`](https://politicaargentina.r-universe.dev/polArViz) to facilitate the visualization of the other packages, and a metapackage called [`{polArverse}`](https://politicaargentina.r-universe.dev/polArverse)that emulates the operation of `{tidyverse}` for loading the set of associated libraries.

Reviewer: ```suggestion
In addition, we worked on an auxiliary library called [polarViz](https://politicaargentina.r-universe.dev/polArViz) to facilitate the visualization of the other packages, and a metapackage called [polArverse](https://politicaargentina.r-universe.dev/polArverse) that emulates the operation of tidyverse for loading the set of associated libraries.
```

> ## 🌠 Meeting of Universes: [the *polArverse* at *R-Universe*](https://politicaargentina.r-universe.dev/builds)

Reviewer: ```suggestion
## 🌠 Meeting of universes: [the *polArverse* and *R-Universe*](https://politicaargentina.r-universe.dev/builds)
```

> During this process, we incorporated R-Universe into the workflow. It provides an autonomous and self-managed space in which to publish these new packages (no longer simply as Github repositories) and provides support for testing on different operating systems and software versions. Undoubtedly, it is a relevant tool for better maintenance.

Reviewer: ```suggestion
During this process, we incorporated the R-Universe into the workflow. It provides an autonomous and self-managed space in which to publish these new packages (no longer simply as GitHub repositories) and provides support for testing on different operating systems and software versions. Undoubtedly, it is an important tool for better maintenance.
```

![](https://lh4.googleusercontent.com/VHMVype__IRMsC3OkW3SHS68Htr8eJD58TPfZpCkycy6iReA_sDjbasHGqwP7RptmvPoD1C6KIYW3RUEEh1NjIxDJNRpmgTdebOz6n1Nzgpc3jLGN8DHYVbkUmYxYGISTc80HuCS-Y3rDl0BtahstFc)

### We continue to take important steps for our universe:

> First, we returned to CRAN! In February 2023, we shipped `{geoAr}, which, as noted, provides access to and facilitates the processing of spatially relevant data. A version under development (accessible from R-Universe) also allows you to interact with the [API of the Argentine Geographic Data Standardization Service (Servicio de Normalización de Datos Geográficos de Argentina](https://georef-ar-api.readthedocs.io/es/latest/).

Reviewer: ```suggestion
First, we returned to CRAN! In February 2023, we shipped geoAr, which, as noted, provides access to and facilitates the processing of spatially relevant data. A version under development (accessible from the R-Universe) also allows you to interact with the [API of the Argentine Geographic Data Standardization Service (Servicio de Normalización de Datos Geográficos de Argentina](https://georef-ar-api.readthedocs.io/es/latest/).
```

On the other hand, we added a pending tool to the polAr toolbox to work with census data: [`{censAr}`](https://github.com/PoliticaArgentina/censAr). The package, currently under development, is based on the code from [`{censo2017}`](https://docs.ropensci.org/censo2017/) developed by [Mauricio "Pacha" Vargas](https://ropensci.org/author/pach%C3%A1-aka-mauricio-vargas-sep%C3%BAlveda/) and part of the rOpenSci suite.

> ## My Experience as a software Developer with R

Reviewer: ```suggestion
## Summarizing my experiences
```

> - I started working with this language - or dialect, [as a colleague likes to say](https://tuqmano.github.io/CienciaDeDatosCiPol/clases/clase1.html#12) about the antecedent of the `S` from *Bell* in the 1970s - a little fortunately.

Reviewer: Do you mean by chance (almost by accident?)? Or luckily?
```suggestion
- I started working with this language - or dialect, [as one of my colleagues likes to say](https://tuqmano.github.io/CienciaDeDatosCiPol/clases/clase1.html#12) regarding the antecedent of S from *Bell* in the 1970s - a little by chance.
```

- I quickly incorporated it as a work tool, in multiple facets: as an analyst, consultant and teacher.

> - The R community -  in particular all the experience around *LatinR*\- were a determining factor for my experience. It certainly made the process much easier.

Reviewer: ```suggestion
- The R community -  in particular all my experiences with *LatinR* - were a determining factor for my development. It certainly made the process much easier.
```

- All in all, I think it is possible to think of improvements in the process to make CRAN an even better space for the ecosystem. `R` and make the FOSS development experience more enjoyable and predictable.

> - The emergence of the project *r-universe* project to enhance the publishing and discovery experience of packages developed in `R` is invaluable.

Reviewer 2: ```suggestion
- The *R-Universe* project to enhance the publishing and discovery experience of packages developed in `R` is invaluable.
```

> The *polarverse* is a side project, on which I advance in my spare time and without funding. It arose almost playfully, as an experiment to learn how to develop software. We tried successfully to publish in CRAN. We got frustrated and angry. We learned. We added new tools. We kept moving forward. Little by little. "Slowly, but surely.

Reviewer 2: ```suggestion
The *polarverse* is a side project on which I advance in my spare time and without funding. It arose almost playfully as an experiment to learn how to develop software. We successfully published it in CRAN. We got frustrated and angry. We learned. We added new tools. We kept moving forward. Little by little. "Slowly, but steady".
```
